pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -285,6 +291,89 @@ protected void removeColumnV1Indices(String column)
     SegmentColumnarIndexCreator.removeColumnMetadataInfo(_segmentProperties, column);
   }
 
+  /**
+   * Right now the text index is supported on RAW (non-dictionary encoded)
+   * single-value STRING columns. Eventually we will relax the constraints
+   * step by step.
+   * For example, later on user should be able to create text index on
+   * a dictionary encoded STRING column that also has native Pinot's inverted
+   * index. We can also support it on BYTE columns later.
+   * @param column column name
+   * @param indexLoadingConfig index loading config
+   * @param fieldSpec field spec
+   */
+  private void checkUnsupForEnablingTextIndexOnNewColumn(String column, IndexLoadingConfig indexLoadingConfig,
+      FieldSpec fieldSpec) {
+    if (!indexLoadingConfig.getNoDictionaryColumns().contains(column)) {
+      throw new UnsupportedOperationException("Text index is currently not supported on dictionary encoded columns");
+    }
+
+    Set<String> sortedColumns = new HashSet<>(indexLoadingConfig.getSortedColumns());
+    if (sortedColumns.contains(column)) {
+      // since Pinot's current implementation doesn't support raw sorted columns,
+      // we need to check for this too
+      throw new UnsupportedOperationException("Text index is currently not supported on sorted columns");
+    }
+
+    if (!fieldSpec.isSingleValueField()) {
+      throw new UnsupportedOperationException("Text index is currently not supported on multi-value columns");
+    }
+
+    if (fieldSpec.getDataType() != FieldSpec.DataType.STRING) {
+      throw new UnsupportedOperationException("Text index is currently only supported on STRING columns");
+    }
+  }
+
+  void createRawForwardIndexForTextIndexEnabledColumn(String column, File segmentIndexDir,
+      SegmentDirectory.Writer segmentWriter, IndexLoadingConfig indexLoadingConfig)
+      throws IOException {
+    FieldSpec fieldSpec = _schema.getFieldSpecFor(column);
+    Preconditions.checkNotNull(fieldSpec);
+    checkUnsupForEnablingTextIndexOnNewColumn(column, indexLoadingConfig, fieldSpec);
```

Comment:
I think make this a Preconditions statement with a booleanm return, and include column namne in the exception is easier.

Reply:
Yes but that way we won't capture the reason (which unsupported operation it it). I included the column name in exception